### DISCUSSION TITLE
I want to display an image in an MD game using Lua. How can I do this?

### DISCUSSION THREAD

**Discussion by uname1able** [mamedev/discussions#146] [Q&A] [answered]
my code ,
        scr = manager.machine.screens[":megadriv"]
        local alpha = 255
        local red = 255
        local green = 0
        local blue = 0
        
        local red_color = (alpha << 24) | (red << 16) | (green << 8) | blue
        for x = 0, 256 - 1 do
            for y = 0, 256 - 1 do
                scr:draw_line(x, y, x+0.5, y,red_color) -- 0xFFFF0000 -- **this color  data will get from pic data matrix** 
            end 
        end 
        
**i want  implent  read ./1.png   -> scr:draw??(x,y,x0,y0)  show pics** 

how can i do this,please give me some code or suggestions, thank you very much!

**Comment by cuavas** [comment, marked as answer]
* Read the file into a [bitmap](https://docs.mamedev.org/luascript/ref-render.html#bitmap) using `emu.bitmap_argb32.load(...)`.
* Create a [texture](https://docs.mamedev.org/luascript/ref-render.html#render-texture) referencing the bitmap using `manager.machine.render:texture_alloc(...)`.
* Use `draw_quad(...)` on a [render container](https://docs.mamedev.org/luascript/ref-render.html#render-container) to draw the texture.

**Reply by uname1able** [reply]
It works,thank you!!!! very much.
